nodeup/pkg/model/kubelet.go

```diff
@@ -441,6 +442,23 @@ func (b *KubeletBuilder) buildKubeletConfigSpec() (*kops.KubeletConfigSpec, erro
 		c.BootstrapKubeconfig = ""
 	}
 
+	if b.Cluster.Spec.Networking != nil && b.Cluster.Spec.Networking.AmazonVPC != nil {
+		instanceType, err := awsup.GetMachineTypeInfo(b.InstanceGroup.Spec.MachineType)
+		if err != nil {
+			return c, err
+		}
+
+		maxPods := int32(instanceType.MaxPods)
+		c.MaxPods = &maxPods
```

Comment:
It looks like all the instance types have MaxPods set, but we might want to check that MaxPods is set (i.e. `!= 0`)

Reply:
And here also, I wouldn't set kubelet maxPods to be >= 110 (which I believe is the default https://github.com/kubernetes/kubernetes/blob/16d099253466eb18fddcab9f9adf6214fe4077ad/pkg/kubelet/apis/config/v1beta1/defaults.go#L159 ).  Most of the network providers can go to > 110 pods, but they don't because other things e.g. docker is the limiting factor.

If the user specifies MaxPods on the Kubelet args explicitly though, I _would_ honor that value - that's a clear statement that the user e.g. is running a container implementation that is more efficient.  In that case I would set the value to `min(kubelet.maxPods, instanceType.maxPods)` (more explanation below)